slick/src/test/scala/akka/stream/alpakka/slick/scaladsl/DocSnippets.scala

```diff
@@ -184,3 +184,72 @@ object SlickFlowExample extends App {
   }
 }
 //#flow-example
+
+//#flowWithPassThrough-example
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+
+import akka.stream.scaladsl._
+import akka.stream.alpakka.slick.scaladsl._
+
+// We're going to pretend we got messages from kafka.
+// After we've written them to a db with Slick, we want
+// to commit the offset to Kafka
+object SlickFlowWithPassThroughExample extends App {
+
+  // mimics a Kafka 'Committable' type
+  case class CommittableOffset(offset: Int) {
+    def commit: Future[Done] = Future.successful(Done)
+  }
+  case class KafkaMessage[A](msg: A, offset: CommittableOffset) {
+    // map the msg and keep the offset
+    def map[B](f: A => B): KafkaMessage[B] = KafkaMessage(f(msg), offset)
+  }
+
+  implicit val system = ActorSystem()
+  implicit val mat = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  implicit val session = SlickSession.forConfig("slick-h2")
+
+  // The example domain
+  case class User(id: Int, name: String)
+  val users = (1 to 42).map(i => User(i, s"Name$i"))
+  val messagesFromKafka = users.zipWithIndex.map { case (user, index) => KafkaMessage(user, CommittableOffset(index)) }
+
+  // This import enables the use of the Slick sql"...",
+  // sqlu"...", and sqlt"..." String interpolators.
+  // See "http://slick.lightbend.com/doc/3.2.1/sql.html#string-interpolation"
+  import session.profile.api._
+
+  // Stream the users into the database as insert statements
+  val done: Future[Done] =
+    Source(messagesFromKafka)
+      .via(
+        // add an optional first argument to specify the parallism factor (Int)
+        Slick.flowWithPassThrough { kafkaMessage =>
+          val user = kafkaMessage.msg
+          (sqlu"INSERT INTO ALPAKKA_SLICK_SCALADSL_TEST_USERS VALUES(${user.id}, ${user.name})")
+            .map { insertCount => // map db result to something else
+              // allows to keep the kafka message offset so it can be committed in a next stage
+              kafkaMessage.map(user => (user, insertCount))
+            }
+        }
+      )
+      .log("nr-of-updated-rows")
+      .mapAsync(1) { // in correct order
+        kafkaMessage =>
+          kafkaMessage.offset.commit // commit kafka messages
+      }
+      .runWith(Sink.ignore)
+
+  done.onComplete {
+    case _ =>
+      session.close()
+      system.terminate()
+  }
+}
+//#flowWithPassThrough-example
```

Comment:
I'd rather see this in a separate file as well just like the Java sample.

Reply:
Looking at the Alpakka sources, it is quite common to have multiple Scala examples in the same file. So since this is according to the current 'standard' I'd like to keep it here. Having them in the same file is convenient IMHO since it allows you easily to compare examples.

Fine by me if you'd like to change this in the future, but I do not want to shave that yak now.

Reply:
Sure, not super important.